worker/caasapplicationprovisioner/application.go

```diff
@@ -236,13 +243,15 @@ func (a *appWorker) loop() error {
 				}
 				appProvisionChanges = appProvisionWatcher.Changes()
 			}
-			err = a.ops.AppAlive(a.name, app, a.password, &a.lastApplied, a.facade, a.clock, a.logger)
-			if errors.Is(err, errors.NotProvisioned) {
-				// State not ready for this application to be provisioned yet.
-				// Usually because the charm has not yet been downloaded.
-				break
-			} else if err != nil {
-				return errors.Trace(err)
+			if !a.statusOnly {
```

Comment:
Would these checks be better as `if a.statusOnly { break } `  to avoid the extra identation etc